lib/phubme/nicknames_matcher.ex

```diff
@@ -1,31 +1,27 @@
 defmodule PhubMe.NicknamesMatcher do
   def match_nicknames({full_comment, github_nicknames, sender, comment_parsed}) do
-    {:ok, full_comment, matching_nicknames(github_nicknames, [], 0), sender, comment_parsed }
+    {:ok, full_comment, matching_nicknames(github_nicknames), sender, comment_parsed }
   end
 
   def match_nicknames(_) do
     {:no_nicknames_found}
   end
 
-  defp matching_nicknames([github_nick_head | github_nick_tail], matched_nicknames, array_position) do
-    updated_matched_nicknames =
-      if nickname_from_mix_config(github_nick_head) == :error do
-        IO.puts "No matching nickname found for " <> List.first(github_nick_head)
-        matched_nicknames
-      else
-        List.insert_at(matched_nicknames, array_position, (github_nick_head ++ nickname_from_mix_config(github_nick_head)))
-      end
-    matching_nicknames(github_nick_tail, updated_matched_nicknames, array_position + 1)
+  defp matching_nicknames([ [ nickname ] | tail]) do
+    case nickname_from_mix_config(nickname) do
+      {:ok, matching_nickname} ->
+        [ [ nickname, matching_nickname ] | matching_nicknames(tail) ]
```

Comment:
The list is build very easily with this syntax. The recursion is not so efficient because the `|/2` operator need both operator to give a result back. Maybe the BEAM VM is smart and figure this out, I don't know. This is still acceptable unless you have a **very** long list of nicknames.